contracts/Vault.sol

```diff
@@ -36,9 +37,11 @@ contract Vault is ERC4626Metadata, Ownable {
 		return owner();
 	}
 
-    function setMarket(address market, uint256 max) public onlyOwner {
+    function setMarket(address market, uint256 max, uint256 rate) public onlyOwner {
         require(_market == address(0), "setMarket: Market already set");
+        require(rate >= 100_000, "setMarket: Rate must be greater than 100,000");
```

Comment:
"100k wei" would be more accurate imo

Reply:
As discussed, this int wie its because no floating point math.